### PR TITLE
Giphy picker 1.0

### PR DESCRIPTION
**Pull request by maartenvdh**
### Motivation

New custom element similar to the Flickr image picker only in this case with the Giphy API.

### Checklist

- [x] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [x] Tests are passing
- [x] Docs have been updated (if applicable)
- [x] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If you want to use the Giphy picker in your project in Kentico Cloud, follow these steps:

* In Kentico Cloud open the Content types tab
* Create or edit a content model to which you want to add the Giphy picker
* Add the **Custom element** content element
* Configure the content element
* Use the following URL as Hosted code URL (HTTPS): https://kentico.github.io/custom-element-samples/GiphyPicker/giphypicker-input-component.html
* Get API key from Giphy: https://developers.giphy.com/
* Provide JSON parameters for the custom element to configure the API Key:

```json
{
    "apikey": "[YOUR API KEY]"
}
```

### CODE REVIEW

File: GiphyPicker/giphypicker-input-component.html [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                try {
+                    $.each(data.data, function (i, item) {
+                        var giphyURL = item.images.fixed_height.url;
+                        img = `<div class='gallery'><img class='giphyimg' src='${giphyURL}'/>`;
+                        $(img).appendTo("#GiphyImages");
+                    });
+                } catch (err) {
+                    // Initialization with Kentico Custom element API failed (page displayed outside of the Kentico UI)
+                    console.error(err);
+                    $("<div class='error'><strong>Images cannot be loaded</strong><br/>Check if you're API key and secret are correct.</div>").appendTo("#flickrImages");
```
Comment (Simply007) [inline GiphyPicker/giphypicker-input-component.html]:
```suggestion
                    $("<div class='error'><strong>Images cannot be loaded</strong><br/>Check if you're API key and secret are correct.</div>").appendTo("#GiphyImages");
```

File: GiphyPicker/giphypicker-input-component.html [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+                $(".giphyimg").click(function (e) {
+                    var fileName = $(e.target).closest("img").attr('src');
+                    if (fileName) {
+                        CustomElement.setValue(fileName);
+                        $("#GiphyImages").html($("<img>").attr("src", fileName));
+                        $("#search").val(fileName);
+                    }
+                    updateSize();
+                });
+            });
```
Comment (Simply007) [inline GiphyPicker/giphypicker-input-component.html]:
```suggestion
            }).fail(function () {
                console.error("error");
                $("<div class='error'><strong>Images cannot be loaded</strong><br/>Check if you're API key and secret are correct.</div>").appendTo("#GiphyImages");
                updateSize();
            });
```

File: GiphyPicker/README.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+limit (optional) number  
+The number of image results to return in the response. The actual number delivered may be less than requested.  
+  
+offset (optional) number  
+The zero-based offset that indicates the number of image results to skip before returning results.  
+  
+rating (optional) string  
+Filter result by rating, possible options are : Y, G, PG, PG-13, R
+  
+lang(optional) string
```
Comment (Simply007) [inline GiphyPicker/README.md]: could you provide the format of the input as an example or link to the docs?

Comment (maartenvdh) [reply GiphyPicker/README.md]: done

**Pull request merged**